modules/wallet/encrypt.go

```diff
@@ -311,23 +296,67 @@ func (w *Wallet) Encrypted() bool {
 	return w.encrypted
 }
 
-// Encrypt will encrypt the wallet using the input key. Upon encryption, a
-// primary seed will be created for the wallet (no seed exists prior to this
-// point). If the key is blank, then the hash of the seed that is generated
-// will be used as the key. The wallet will still be locked after encryption.
+// Encrypt will create a primary seed for the wallet and encrypt it using
+// masterKey. If masterKey is blank, then the hash of the primary seed will be
+// used instead. The wallet will still be locked after Encrypt is called.
 //
 // Encrypt can only be called once throughout the life of the wallet, and will
 // return an error on subsequent calls (even after restarting the wallet). To
-// reset the wallet, the wallet files must be moved to a different directory or
-// deleted.
+// reset the wallet, the wallet files must be moved to a different directory
+// or deleted.
 func (w *Wallet) Encrypt(masterKey crypto.TwofishKey) (modules.Seed, error) {
 	if err := w.tg.Add(); err != nil {
 		return modules.Seed{}, err
 	}
 	defer w.tg.Done()
 	w.mu.Lock()
 	defer w.mu.Unlock()
-	return w.initEncryption(masterKey)
+
+	// Create a random seed.
+	var seed modules.Seed
+	_, err := rand.Read(seed[:])
+	if err != nil {
+		return modules.Seed{}, err
+	}
+
+	// If masterKey is blank, use the hash of the seed.
+	if masterKey == (crypto.TwofishKey{}) {
+		masterKey = crypto.TwofishKey(crypto.HashObject(seed))
+	}
+
+	return w.initEncryption(masterKey, seed)
+}
+
+// InitFromSeed functions like Init, but using a specified seed. Unlike Init,
+// the blockchain will be scanned to determine the seed's progress. For this
+// reason, InitFromSeed should not be called until the blockchain is fully
+// synced.
+func (w *Wallet) InitFromSeed(masterKey crypto.TwofishKey, seed modules.Seed) error {
+	if err := w.tg.Add(); err != nil {
+		return err
+	}
+	defer w.tg.Done()
+
+	if !w.cs.Synced() {
+		return errors.New("cannot init from seed until blockchain is synced")
+	}
```

Comment:
that is precisely what I do, haha. `InitFromSeed` creates a `seedScanner`, scans the blockchain, and then uses `s.largestIndexSeen`. The `seedScanner` will still accumulate the outputs, but `InitFromSeed` ignores them.